pkg/crc/api/api.go

```diff
@@ -91,21 +100,28 @@ func (api CrcAPIServer) handleConnections(conn net.Conn) {
 	// start, stop and delete are slow operations, and change the VM state so they have to run sequentially.
 	// We don't want other operations querying the status of the VM to be blocked by these,
 	// so they are treated by a dedicated go routine
-	if req.Command == "start" || req.Command == "stop" || req.Command == "delete" {
+
+	switch req.Command {
+	case "start", "stop", "delete":
 		// queue new request to channel
 		r := clusterOpsRequest{
 			command: req,
 			socket:  conn,
 		}
 		if !addRequestToChannel(r, api.clusterOpsRequestsChan) {
-			defer conn.Close()
 			logging.Error("Channel capacity reached, unable to add new request")
 			errMsg := encodeErrorToJSON("Sockets channel capacity reached, unable to add new request")
 			writeStringToSocket(conn, errMsg)
-			return
+			conn.Close()
 		}
-	} else {
+
+	case "status", "version", "setconfig", "getconfig", "unsetconfig", "webconsoleurl":
 		go api.handleRequest(req, conn)
+
+	default:
+		err := encodeErrorToJSON(fmt.Sprintf("Unknown command supplied: %s", req.Command))
+		writeStringToSocket(conn, err)
+		conn.Close()
 	}
```

Comment:
I don't think this change in `handleConnections` is related to the handler interface change? Maybe indirectly because you moved the "unknown command" reporting here rathre than in `handleRequest`?

Reply:
Yeah indirectly, there are two places where commands are filtered, and i went with different message, actually both should report "Unknown command"